foundationdb/src/transaction.rs

```diff
@@ -464,6 +464,51 @@ impl Transaction {
     fn new_future(&self, f: *mut fdb::FDBFuture) -> TrxFuture {
         TrxFuture::new(self.clone(), f)
     }
+
+    /// Returns an FDBFuture which will be set to the versionstamp which was used by any
+    /// versionstamp operations in this transaction. You must first wait for the FDBFuture to be
+    /// ready, check for errors, call fdb_future_get_key() to extract the key, and then destroy the
+    /// FDBFuture with fdb_future_destroy().
+    ///
+    /// The future will be ready only after the successful completion of a call to
+    /// fdb_transaction_commit() on this Transaction. Read-only transactions do not modify the
+    /// database when committed and will result in the future completing with an error.  Keep in
+    /// mind that a transaction which reads keys and then sets them to their current values may be
+    /// optimized to a read-only transaction.
+    ///
+    /// Most applications will not call this function.
+    pub fn get_versionstamp(&self) -> TrxVersionstamp {
+        let trx = self.inner.inner;
+
+        let f = unsafe { fdb::fdb_transaction_get_versionstamp(trx) };
+        TrxVersionstamp {
+            inner: FdbFuture::new(f),
+        }
+    }
+
+    /// The transaction obtains a snapshot read version automatically at the time of the first call
+    /// to fdb_transaction_get_*() (including this one) and (unless causal consistency has been
+    /// deliberately compromised by transaction options) is guaranteed to represent all
+    /// transactions which were reported committed before that call.
+    pub fn get_read_version(&self) -> TrxReadVersion {
```

Comment:
I haven’t read the docs enough to understand this yet. Maybe you can explain this to me... what is the difference between this version and the VersionStamp?

I.E. why is this i64 and in the other context a large set of 10 or 12 bytes?

Reply:
It seems that a versionstamp consists of 8-byte long transaction version and 2-byte transaction batch order. `get_read_version` returns transcation version only, which increments roughly one per each microsecond (1,000,000 hz) [1]

[1] https://forums.foundationdb.org/t/version-length-is-53-bits-enough/260/5

Reply:
That link didn’t explain a ton. So negative number means what? Or is u64 just as valid?